winter/http/media_type.py

```diff
@@ -53,35 +53,48 @@ def subtype(self) -> str:
     def parameters(self) -> Mapping[str, str]:
         return self._parameters
 
-    @staticmethod
-    def parse(media_type: str) -> Tuple[str, str, Mapping[str, str]]:
+    @classmethod
+    def parse(cls, media_type: str) -> Tuple[str, str, Mapping[str, str]]:
         media_type = media_type.strip()
         if not media_type:
             raise InvalidMediaTypeException(media_type, 'Media type must not be empty')
+
+        full_type, *parameters_parts = media_type.split(';')
+        parameters = cls._build_parameters(media_type, parameters_parts)
+        type_, separator, subtype = full_type.partition('/')
```

Comment:
I suggest to extract it to a method (for example `type_, subtype = _parse_full_type(full_type)`) and validate it there completely. raise `Media type must contain "/"` from it and other related checks. It will simplify the following code.

Reply:
Done